### DISCUSSION TITLE
Thread safety

### DISCUSSION THREAD

**Discussion by jvmlet** [ZiggyCreatures/FusionCache#297] [General]
Hi
The reason I'm looking for alternatives to donet default cache is [this](https://blog.novanet.no/asp-net-core-memory-cache-is-get-or-create-thread-safe/)
Given that `FusionCache` relies  on `MemoryCache`, how the thread safety issue has been addressed ?

**Comment by jodydonetti** [comment]
Hi @jvmlet , yes!
What you are referring to is known as the Cache Stampede problem, and you can read more about it [here](https://github.com/ZiggyCreatures/FusionCache/blob/main/docs/CacheStampede.md).

Long story short: by using FusionCache instead of MemoryCache this is automatically solved.

ps: even if you'll add in the future a distributed cache as an L2 (see [here](https://github.com/ZiggyCreatures/FusionCache/blob/main/docs/CacheLevels.md)), the cache stampede protection will also work there.

Hope this helps!

**Reply by jvmlet** [reply]
Greate, thanks.

**Reply by jodydonetti** [reply]
You're welcome!

If you can, please mark it as the answer, thanks.

**Reply by jvmlet** [reply]
I can't,  https://github.com/orgs/community/discussions/2903

**Reply by jodydonetti** [reply]
Ah ok, got it!
